### PR TITLE
Angular12

### PR DESCRIPTION
**Pull request by KristianEkman**
I'm not sure if I've updated all readmes and such.
And it is not all fully tested on Angular 9-12.
And, do you see this as a minor upgrade --> 3.5.8?

### CODE REVIEW

**Comment by jaibatrik** [comment]
That's not a problem. I think we can consider this as a minor release, not patch. If this is not compatible with 9 / 10 / 11 versions of Angular, we can even call it a major release I think. I'm not working with Angular that much lately. Do you know if there is a way of testing if it works with previous Angular versions?

**Comment by jaibatrik** [comment]
Also, we can probably transition to using Ivy at this point - https://angular.io/guide/creating-libraries#transitioning-libraries-to-partial-ivy-format

**Comment by KristianEkman** [comment]
It is quite common to increment major version when upgrading Angular
dependencies.
It might work with earlier releases also but it takes time to test that.
I guess you have to down-grade the angual cli, create a new app and add
this new lib to it.
Then compile and run it.

Den lör 26 juni 2021 kl 14:28 skrev Jaibatrik Dutta <
***@***.***>:

**Comment by jaibatrik** [comment]
Great, we can go with major upgrade in that case. Would you be able to change the version in `package.json` in that case?

Thanks again for the contribution!

File: projects/lib/src/providers/dummy-login-provider.ts [hunk 1]
```diff
@@ -90,7 +88,7 @@ export class DummyLoginProvider extends BaseLoginProvider {
   signOut(revoke?: boolean): Promise<any> {
     return new Promise((resolve, reject) => {
       this.loggedIn = false;
-      resolve();
+      resolve(this.dummy);
```
Comment (jaibatrik) [inline projects/lib/src/providers/dummy-login-provider.ts:91]: Could you please revert this code? I'll fix it in a different PR where I change the method signature.

File: projects/lib/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angularx-social-login",
-  "version": "3.5.7",
```
Comment (jaibatrik) [inline projects/lib/package.json:3]: I think you can revert the changes in this file as well. Let me make the changes after testing and before publishing to npm.

**Review by jaibatrik — CHANGES_REQUESTED** [review]
Thanks @KristianEkman. I have requested couple of minor changes. Rest of it looks good to me!

**Comment by KristianEkman** [comment]
Sure. Ill also test that my Online Backgammon site works as expected before I push the last commits.

File: projects/lib/src/providers/dummy-login-provider.ts (hunk 1 above)
Comment (KristianEkman) [reply projects/lib/src/providers/dummy-login-provider.ts:91]: ok

File: projects/lib/package.json (hunk 2 above)
Comment (KristianEkman) [reply projects/lib/package.json:3]: ok

**Comment by KristianEkman** [comment]
Only been able to test google provider. But I think the others will work fine.

**Review by jaibatrik — APPROVED** [review]
